sdk/search/azure-search-documents/azure/search/documents/_index/_search_index_client.py

```diff
@@ -110,6 +102,7 @@ def get_document_count(self, **kwargs):
 
         :rtype: int
         """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
```

Comment:
Whenever we're extracting specific named vars from kwargs it makes me wonder if we could be using named args instead to make the optional logic more clear?  (or at the very least, if headers should exist as a docstring)  This may not be feasible depending on how kwargs is being used downstream and/or we may want to deemphasize header population except via advanced users, so there could be good reasons not to, but it's a thought that comes to mind.

Reply:
I definitely thought about it. I think I have a slight preference for passing the named arg but it does not seem to be the common practice in this repo